core/src/main/java/edu/wpi/grip/core/serialization/Project.java

```diff
@@ -127,19 +130,42 @@ public void save(File file) throws IOException {
 
   public void save(Writer writer) {
     this.xstream.toXML(this.pipeline, writer);
-    saveIsDirty = false;
+    saveIsDirty.set(false);
   }
 
   public boolean isSaveDirty() {
-    return saveIsDirty;
+    return saveIsDirty.get();
+  }
+
+  public void addIsSaveDirtyConsumer(Consumer<Boolean> consumer) {
+    saveIsDirty.addConsumer(consumer);
   }
 
   @Subscribe
   public void onDirtiesSaveEvent(DirtiesSaveEvent dirtySaveEvent) {
     // Only update the flag the save isn't already dirty
     // We don't need to be redundantly checking if the event dirties the save
-    if (!saveIsDirty && dirtySaveEvent.doesDirtySave()) {
-      saveIsDirty = true;
+    if (!saveIsDirty.get() && dirtySaveEvent.doesDirtySave()) {
+      saveIsDirty.set(true);
+    }
+  }
+
+  public class ObservableBoolean {
```

Comment:
Does this need to be public? Should also make this static, does look like it needs a reference to the Project instance